### PR TITLE
do_meson_build: use cross-file

### PR DESCRIPTION
**Pull request by kencu**
this allows cross arch building with meson
eg building arm64 on Intel

### CODE REVIEW

**Comment by kencu** [comment]
For meson:

the build machine is the machine doing the building.

the host machine is the machine (arch) that the products will run on.

the target machine is the machine that any software built by the host machine would run on (not relevant here).

So the cross file has to spec out the host machine. Meson auto-detects the build machine (although we pass in the CC and CXX_FOR_BUILD so it uses what we want.

The target machine is not relevant here, as we're not building any compilers that will run on the host machine.


So -- say all that three times backwards and pat your head :>

**Comment by kencu** [comment]
On MacPorts I reinplace the toolchain path when installing the cross files. 

I'm not quite sure how you want to handle that in this case, so I just hard-coded them to the build dir. Perhaps there is some way to reinplace the path in shell?

Eg:

```
pkgconfig = '/opt/buildX11/bin/pkg-config'
```

File: compile.sh
Comment (jeremyhu) [inline compile.sh:123]: nit: Remove the extra line of whitespace here.

**Review by jeremyhu — APPROVED** [review]

**Comment by jeremyhu** [comment]
Thanks, @kencu.  Go ahead and merge this once you fix that whitespace nitpick.

**Comment by kencu** [comment]
nit picked :>

But I canna commit 'er -- she just wonna lemme...

**Comment by jeremyhu** [comment]
> nit picked :>
> 
> But I have no commit rights at present.

I just sent you an invite to have write access.  Once you accept, you should have write access and can merge.

Thanks.

**Comment by kencu** [comment]
Thanks!

**Pull request merged**
